crates/cast/src/starknet_commands/declare.rs

```diff
@@ -72,14 +73,18 @@ pub async fn declare(
     let result_code = command_result
         .status
         .code()
-        .context("failed to obtain status code from scarb build")?;
+        .context("Failed to obtain status code from scarb build")?;
+    let result_msg = String::from_utf8(command_result.stdout)?;
     if result_code != 0 {
-        anyhow::bail!("scarb build returned non-zero exit code: {}", result_code);
+        anyhow::bail!(
+            "Scarb build returned non-zero exit code: {} - error message: {}",
+            result_code,
+            result_msg
+        );
     }
 
     // TODO #41 improve handling starknet artifacts
-    let compiled_directory = find_manifest_path(None)
-        .expect("Failed to obtain Scarb.toml file path")
+    let compiled_directory = &manifest_path
         .parent()
         .map(|parent| parent.join("target/release"))
```

Comment:
This should read target dir from`scarb metadata` instead.

Reply:
I suppose it should:

1. Run Command::new("scarb").arg("metadata") (...)
2. get stdout out of it
3. transform to String
4. Transform to serialize to json
5. get target_dir
6. append "/release" to it

?

Or is there any other, simpler way? tried to check scarb_metadata, but couldn't find the thing I needed.